### PR TITLE
Fetch table definitions for TSQL

### LINKED ISSUE #918: [FEATURE]: Fetch TableDefinition entries from all catalogs/databases/tables in TSQL
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Category of feature request

Transpile

### Problem statement

 Fetch TableDefinition entries from all catalogs/databases/tables in TSQL.

### Proposed Solution

Implementation is done for Snowflake https://github.com/databrickslabs/remorph/issues/902. 
The same thing will be implemented in TSQL.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by sriram251-code**
closes #918 

This pull request retrieves table definitions from TSQL and structures them using a case class named TableDefinition.

The TableDefinition class encapsulates the following properties for each table:

catalog: The catalog name (String)
schema: The schema name (String)
table: The table name (String)
location: An optional field for table location (Option[String])
tableFormat: An optional field for the table format (Option[String])
viewText: An optional field for view definition, if applicable (Option[String])
columns: A sequence of ColumnDetail objects representing the table's columns (Seq[ColumnDetail])
sizeGb: The table size in gigabytes (Int)
comments: The comment given on the table(Option[String])

This structured approach allows for efficient querying and manipulation of Snowflake table metadata within the application.

### CODE REVIEW

File: core/src/main/scala/com/databricks/labs/remorph/discovery/queries.scala [hunk 1]
```diff
@@ -19,7 +19,8 @@ case class TableDefinition(
     tableFormat: Option[String] = None,
     viewText: Option[String] = None,
     columns: Seq[StructField] = Seq.empty,
-    sizeGb: Int = 0)
+    sizeGb: Int = 0,
+    comments: Option[String] = None)
```
Comment (sundarshankar89) [inline core/src/main/scala/com/databricks/labs/remorph/discovery/queries.scala:23]: Just checking is the ask is to capture only Table Comments, how about Column Comments?

Comment (sriram251-code) [reply core/src/main/scala/com/databricks/labs/remorph/discovery/queries.scala:23]: @nfx @gueniai let us know if we need to capture the column comments? Now we are only capturing table comments.

Comment (gueniai) [reply core/src/main/scala/com/databricks/labs/remorph/discovery/queries.scala:23]: We discussed this during standup, but just confirming here that we do need column comments as well

File: core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/ddl.scala [hunk 2]
```diff
@@ -59,6 +59,7 @@ case class StructField(name: String, dataType: DataType, nullable: Boolean = tru
 case class StructType(fields: Seq[StructField]) extends DataType
 case class MapType(keyType: DataType, valueType: DataType) extends DataType
 case object VariantType extends DataType
+case class ColumnDetail(name: String, dataType: DataType, nullable: Boolean = true, comment: Option[String] = None)
```
Comment (sundarshankar89) [inline core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/ddl.scala:62]: Can you add a comment on what or why this IR is needed

Comment (sriram251-code) [reply core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/ddl.scala:62]: As we can't use StructField to store the comments for each column

File: core/src/main/scala/com/databricks/labs/remorph/discovery/queries.scala (hunk 1 above)
Comment (sriram251-code) [reply core/src/main/scala/com/databricks/labs/remorph/discovery/queries.scala:23]: done

**Pull request closed without merging**